components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/RedisSessionDataStore.java

```diff
@@ -0,0 +1,415 @@
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingDeque;
+
+public class RedisSessionDataStore extends SessionDataStore {
+
+    public static final String NANO_TIME = "nanoTime";
+    public static final String TYPE = "type";
+    public static final String EXPIRY_TIME = "expireTime";
+    public static final String TENANT_ID = "tenantId";
+    public static final String HOST = "127.0.0.1";
+    public static final int PORT = 6379;
+    public static final String OBJECT = "object";
+    public static final String DIVIDER = ":";
+    public static final int MAX_TOTAL = 1000;
+    public static final int MAX_IDLE = 1000;
+    public static final int MIN_IDLE = 100;
+    public static final int MAX_WAIT_MILLIS = 20000;
+    private static final int DEFAULT_DELETE_LIMIT = 50000;
+    private static final Log log = LogFactory.getLog(SessionDataStore.class);
+    private static final BlockingDeque<SessionContextDO> tempAuthnContextDataDeleteQueue = new LinkedBlockingDeque();
+    private static final BlockingDeque<SessionContextDO> sessionContextQueue = new LinkedBlockingDeque();
+    private static boolean tempDataCleanupEnabled = false;
+    private static int maxSessionDataPoolSize = 100;
+    private static int maxTempDataPoolSize = 50;
+    private static String sqlDeleteExpiredDataTask = "";
+    private static JedisPool pool;
+
+    static {
+        try {
+            String maxPoolSizeValue = IdentityUtil.getProperty("JDBCPersistenceManager.SessionDataPersist.PoolSize");
+            if (StringUtils.isNotBlank(maxPoolSizeValue)) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Session data pool size config value: " + maxPoolSizeValue);
+                }
+                maxSessionDataPoolSize = Integer.parseInt(maxPoolSizeValue);
+            }
+
+            String isTempDataCleanupEnabledVal = IdentityUtil.getProperty("JDBCPersistenceManager.SessionDataPersist.TempDataCleanup.Enable");
+            if (StringUtils.isNotBlank(isTempDataCleanupEnabledVal)) {
+                tempDataCleanupEnabled = Boolean.parseBoolean(isTempDataCleanupEnabledVal);
+            }
+
+            String maxTempDataPoolSizeValue = IdentityUtil.getProperty("JDBCPersistenceManager.SessionDataPersist.TempDataCleanup.PoolSize");
+            if (StringUtils.isNotBlank(maxTempDataPoolSizeValue)) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Temporary data pool size config value: " + maxPoolSizeValue);
+                }
+                maxTempDataPoolSize = Integer.parseInt(maxTempDataPoolSizeValue);
+            }
+
+        } catch (NumberFormatException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Exception ignored : ", e);
+            }
+            log.warn("One or more pool size configurations cause NumberFormatException. Default values would be used");
```

Comment:
We may improve the debug log message to something similar to `Error occurred while parsing the configurations`

Reply:
Also in the warn log, we can omit the developer level details as sometimes a sys-admin might see this error and they may not be familiar with dev level errors. We can slightly modify the above log to `"An error occurred while parsing one or more pool size configurations. Default values would be used"`